### ISSUE TITLE
Dot navigation breaks when carousel has less items than set

### ISSUE THREAD

**Issue by benhodgson87** [YIZHUANG/react-multi-carousel#103]
**Describe the bug**
Dot navigation breaks when carousel contains less items than set in the responsive prop.

**To Reproduce**
1. Set slider to show 4 items at once
2. Add 2 items to slider
3. Should show 2 items, aligned to left, with single inactive dot
4. Clicking dot is still possible, which causes items to slide to the right, and sets dot to active

```jsx
showDots={true}
responsive={{
  desktop: {
    breakpoint: { max: 4096, min: 721 },
    items: 4,
    slidesToSlide: 4
  },
  mobile: {
    breakpoint: { max: 720, min: 0 },
    items: 2,
    slidesToSlide: 2
  }
}}
```

![Nov-14-2019 11-57-17](https://user-images.githubusercontent.com/189707/68855343-f2d3f800-06d5-11ea-8c82-2c6e4604bc0a.gif)

**Expected behavior**
If there aren't enough items currently visible at a breakpoint to show more than one slide, the dots should either be hidden as navigation/showing the user where they are is unnecessary, or the button should be disabled.

eg. `<button disabled={carouselState.totalItems <= carouselState.slidesToShow}>`

**Desktop:**
 - OS: OSX
 - Firefox 69
 - Chrome 77

**Comment by benhodgson87** [comment]
Also just tested this with `infinite={true}` set and seeing a completely different bug.

Instead of the dot being inactive, it is active, but clickable. Clicking it causes the slides to completely disappear. Clicking the button again does nothing.

![Nov-14-2019 12-19-21](https://user-images.githubusercontent.com/189707/68856717-206e7080-06d9-11ea-829c-f5d14e341a8e.gif)

**Comment by benhodgson87** [comment]
Resolved by #104

**Comment by YIZHUANG** [comment]
It's now in 2.4.1
